src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4703,7 +4713,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           dyna.wrapErrors(t, (_.typed1(t, mode, pt)))
         }
 
-        val sym = tree.symbol orElse member(qual, name) orElse {
+        val sym = tree.symbol orElse member(qual, name) orElse inCompanionForJavaStatic(qual.tpe.prefix, qual.symbol, name) orElse {
```

Comment:
It is safe to just look at the type+symbol of the `qual` and discard the tree itself because we are only typechecking Java type signatures, not arbitrary Java expressions.